IdentityCore/src/requests/MSIDNonceTokenRequest.h

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+@class MSIDRequestParameters;
+const static NSUInteger kMSIDNonceLifetimeInSeconds = 180;
+
+@interface MSIDCachedNonce : NSObject
+
+@property (nonatomic, readonly, nonnull) NSString *nonce;
+@property (nonatomic, readonly, nonnull) NSDate *cachedDate;
+
+- (_Nonnull instancetype)init NS_UNAVAILABLE;
++ (_Nonnull instancetype)new NS_UNAVAILABLE;
+- (_Nonnull instancetype)initWithNonce:(nonnull NSString *)nonce;
+
+@end
+
+typedef void (^MSIDNonceRequestCompletion)(NSString * _Nullable resultNonce, NSError * _Nullable error);
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSIDNonceTokenRequest : NSObject
```

Comment:
Please create a new h/m for a separate class

Reply:
This is the h/m class for MSIDNonceTokenRequest

Reply:
Broker had it one so had kept the same... Shifted to a new class now